unified_planning/grpc/proto_reader.py

```diff
@@ -0,0 +1,544 @@
+    elif op == "<":
+        return OperatorKind.LT
+    elif op == "and":
+        return OperatorKind.AND
+    elif op == "or":
+        return OperatorKind.OR
+    elif op == "not":
+        return OperatorKind.NOT
+    elif op == "exists":
+        return OperatorKind.EXISTS
+    elif op == "forall":
+        return OperatorKind.FORALL
+    elif op == "implies":
+        return OperatorKind.IMPLIES
+    elif op == "iff":
+        return OperatorKind.IFF
+
+    raise ValueError(f"Unknown operator `{op}`")
+
+
+class ProtobufReader(Converter):
+    @handles(unified_planning_pb2.Parameter)  # type: ignore
+    def _convert_parameter(self, msg, problem):
+        return Parameter(
+            msg.name,
+            convert_type_str(msg.type, problem),
+        )
+
+    @handles(unified_planning_pb2.Fluent)  # type: ignore
+    def _convert_fluent(self, msg, problem):
+        value_type = convert_type_str(msg.value_type, problem)
+        sig = []
+        for p in msg.parameters:
+            sig.append(self.convert(p, problem))
+        fluent = unified_planning.model.Fluent(msg.name, value_type, sig, problem.env)
+        return fluent
+
+    @handles(unified_planning_pb2.ObjectDeclaration)  # type: ignore
+    def _convert_object(self, msg, problem):
+        obj = unified_planning.model.Object(
+            msg.name, convert_type_str(msg.type, problem)
+        )
+        return obj
+
+    @handles(unified_planning_pb2.Expression)  # type: ignore
+    def _convert_expression(self, msg, problem, param_map):
+        if msg.kind == unified_planning_pb2.ExpressionKind.Value("CONSTANT"):
+            assert msg.atom is not None
+            return self.convert(msg.atom, problem)
+
+        elif msg.kind == unified_planning_pb2.ExpressionKind.Value("PARAMETER"):
+            return problem.env.expression_manager.ParameterExp(
+                param=Parameter(
+                    msg.atom.symbol, convert_type_str(msg.type, problem)
+                ),
+            )
+        elif msg.kind == unified_planning_pb2.ExpressionKind.Value("VARIABLE"):
+            return problem.env.expression_manager.VariableExp(
+                var=Variable(msg.atom.symbol, convert_type_str(msg.type, problem)),
+            )
+        elif msg.kind == unified_planning_pb2.ExpressionKind.Value("STATE_VARIABLE"):
+            args = []
+            payload = None
+
+            fluent = msg.list.pop(0)
+            if fluent.kind == unified_planning_pb2.ExpressionKind.Value(
+                "FLUENT_SYMBOL"
+            ):
+                payload = self.convert(fluent.atom, problem)
+            else:
+                args.append(self.convert(fluent, problem, param_map))
+
+            args.extend([self.convert(m, problem, param_map) for m in msg.list])
+            return problem.env.expression_manager.create_node(
+                node_type=OperatorKind.FLUENT_EXP,
+                args=tuple(args),
+                payload=payload,
+            )
```

Comment:
Why did you use the low lovel `create_node` method?

```suggestion
            return problem.env.expression_manager.FluentExp(payload, tuple(args))
```